### PR TITLE
CPDRP-812: validation test scenarios

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
CPDRP-812
Test data for manual testing of participant validation. Documented in https://docs.google.com/document/d/1PlBZQHqSmD3QxYzTpAvd98-rbLzkww-jdync3fYUVv4/edit#

Also changed:
- Use preprod DQT api in pre-live environments (not the broken "dev" environment)
- Make the seeds idempotent

### CODE REVIEW

File: db/seeds/test_data.rb [hunk 1]
```diff
@@ -208,3 +208,102 @@
...
+end
+ParticipantProfile::NPQ.find_or_create_by!(teacher_profile: teacher_profile)
+
+user = User.find_or_create_by!(email: "fip-mentor2@example.com") do |u|
+  u.full_name = "FIP Mentor"
+end
+teacher_profile = TeacherProfile.find_or_create_by!(user: user)
+ParticipantProfile::Mentor.find_or_create_by!(teacher_profile: teacher_profile) do |mentor_profile|
+  mentor_profile.school_cohort = School.find_by(urn: "000105").school_cohorts.find_by(cohort: Cohort.current)
+end
```
Comment (kicferk1) [inline db/seeds/test_data.rb:257]: Soo, are they linked in any way, other than that we know they are the same?

Or is the point of linking when the mentor user enters their trn which is the same into our dqt validation?

**Review by kicferk1 — APPROVED** [review]
Looks pretty good to me, did we write down the emails for testing somewhere prominent so that we don't need to look through the code to know how to test an edge case?

File: db/seeds/test_data.rb (hunk 1 above)
Comment (ethanmills) [reply db/seeds/test_data.rb:257]: They're not linked in any particular way, that's just the email I told people to use in the instructions

**Comment by ethanmills** [comment]
> Looks pretty good to me, did we write down the emails for testing somewhere prominent so that we don't need to look through the code to know how to test an edge case?

Google doc in description 🙂
